conda/core/python_dist.py

```diff
@@ -0,0 +1,444 @@
+                fpath = join(path, fname)
+                if isfile(fpath):
+                    metadata_path = fpath
+                    break
+        elif isfile(path):
+            # '<pkg>.egg-info' file contains metadata directly
+            metadata_path = path
+        else:
+            raise MetadataConflictError
+
+        return metadata_path
+
+    def _read_metadata(self, fpath):
+        """"""
+        if isfile(fpath):
+            if fpath.endswith('.json'):
+                self._data = self._read_json_metadata(fpath)
+            else:
+                self._data = self._parse_metadata(fpath)
+        else:
+            self._data = {}
+
+    def _read_json_metadata(self, fpath):
+        """"""
+        data = None
+        if isfile(fpath):
+            with open(fpath, 'r') as f:
+                data = json.loads(f.read())
+        return data
+
+    def _parsed_metadata_to_json(self, data):
+        """Parse the original format which is stored as RFC-822 headers."""
+        new_data = OrderedDict() if data else None
+        if data:
+            for key, value in data.items():
+                new_key = key.lower().replace('-', '_')
+                if key in METADATA_MULTIPLE_KEYS:
+                    if new_key not in new_data:
+                        new_data[new_key] = [value]
+                    else:
+                        new_data[new_key].append(value)
+                elif key in METADATA_SINGLE_KEYS:
+                    new_data[new_key] = value
+                else:
+                    raise MetadataConflictError
+
+            # TODO: If we need to handle ALL edge cases we might just use the
+            # current metadata.py in distlib, or we could use a trimmed down
+            # version
+
+            # Multiple values should use plural keys?
+            # Handle body?
+            # Handle license?
+
+            # Handle keywords
+            if 'keywords' in new_data:
+                # TODO: Handle comma, semicolon?
+                new_data['keywords'] = new_data['keywords'].split()
+
+        return new_data
+
+    def _parse_metadata(self, fpath):
+        """"""
+        data = None
+        if isfile(fpath):
+            parser = email.parser.HeaderParser()
+            with open(fpath, 'r') as fp:
+                data = parser.parse(fp)
+        return self._parsed_metadata_to_json(data)
+
+    def get_run_requirements(self, exclude_markers=False):
+        """"""
+        requirements = []
+        if self._data:
+            raw_requirements = self._data.get('requires_dist', [])
+            for req in raw_requirements:
+                if exclude_markers:
+                    requirements.append(req.split(';')[0].strip())
+                else:
+                    requirements.append(req.strip())
+        return tuple(requirements)
+
+    @property
+    def name(self):
+        return self._data.get('name')  # TODO: Check for existence?
```

Comment:
Yes, it should be some type of error if the dist doesn't have a name.  I absolute minimum requirements are name and version.  If one is not present, let's assume for now that we're going to emit a warning, and then move on and ignore that package.